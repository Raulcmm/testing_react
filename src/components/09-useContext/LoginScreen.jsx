import React, { useContext } from "react";
import { userContext } from "./userContext";

const LoginScreen = () => {
	const { setUser } = useContext(userContext);
	return (
		<div>
			<h1>LoginScreen</h1>
			<hr />
			<button
				className="btn btn-primary"
				onClick={() => setUser({ id: 123, name: "Osvaldo Camacho" })}
			>
				Agregar datos
			</button>
		</div>
	);
};

export default LoginScreen;
