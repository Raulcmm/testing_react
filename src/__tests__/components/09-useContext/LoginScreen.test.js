import { mount } from "enzyme";
import React from "react";

import LoginScreen from "../../../components/09-useContext/LoginScreen";
import UserContextProvider, {
	userContext,
} from "../../../components/09-useContext/userContext";

describe("Test to Login Screen", () => {
	const setUser = jest.fn();
	const wrapper = mount(
		<userContext.Provider
			value={{
				setUser,
			}}
		>
			<LoginScreen />
		</userContext.Provider>
	);
	test("Match con el snap", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe setear la informacion con el objeto enviado", () => {
		wrapper.find("button").simulate("click");
		expect(setUser).toHaveBeenCalledWith({
			id: 1223,
			name: "Osvaldo Camacho",
		});
	});
});
