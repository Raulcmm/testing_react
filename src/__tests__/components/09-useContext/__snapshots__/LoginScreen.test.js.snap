// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to Login Screen Match con el snap 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Agregar datos
  </button>
</div>
`;
